org-code-javabuilder/theater/src/main/java/org/code/theater/Theater.java

```diff
@@ -0,0 +1,196 @@
+    * @param color  the color to draw the text, using any CSS color string (e.g. #234 or green)
+    * @param font   the name of the font to draw the text in
+    * @param height the height of the text in pixels.
+    */
+   public static void drawText(String text, double x, double y, String color, String font, double height) {
+   }
+
+   /**
+    * Draw a line on the cavas.
+    * 
+    * @param startX the beginning X coordinate of the line.
+    * @param startY the beginning Y coordinate of the line.
+    * @param endX   the end X coordinate of the line.
+    * @param endY   the end Y coordinate of the line.
+    */
+   public static void drawLine(double startX, double startY, double endX, double endY) {
+   }
+
+   /**
+    * Draw a regular polygon on the canvas.
+    * 
+    * @param x     the center X coordinate of the polygon
+    * @param y     the center Y coordinate of the polygon
+    * @param sides the number of sides of the polygon
+    * @param size  the distance from the center to each point on the polyon
+    */
+   public static void drawPolygon(double x, double y, int sides, double size) {
+   }
+
+   /**
+    * Draw as a shape by connecting the points provided.
+    * 
+    * @param points an array of numbers representing the points. For instance, a
+    *               triangle could be represented as [x1, y1, x2, y2, x3, y3].
+    * @param close  whether to close the shape. If this is set to true, the last
+    *               point and the first point will be connected by a line, and if a
+    *               fill color is set, the shape will be filled with that color.
+    */
+   public static void drawShape(double[] points, boolean close) {
+   }
+
+   /**
+    * Draws an ellipse (an oval or a circle) on the canvas.
+    * 
+    * @param x      the left side of the ellipse.
+    * @param y      the top of the ellipse
+    * @param width  the width of the ellipse
+    * @param height the height of the ellipse
+    */
+   public static void drawEllipse(double x, double y, double width, double height) {
+   }
+
+   /**
+    * Draws a rectangle on the canvas.
+    * 
+    * @param x      the left side of the rectangle.
+    * @param y      the top of the rectangle.
+    * @param width  the width of the rectangle.
+    * @param height the height of the rectangle.
+    */
+   public static void drawRectangle(double x, double y, double width, double height) {
+   }
+
+   /**
+    * Sets the thickness of lines drawn
+    * 
+    * @param width
+    */
+   public static void setStrokeWidth(double width) {
+   }
+
+   /**
+    * Sets the color of lines drawn.
+    * 
+    * @param CSS color string 
+    */
+   public static void setStrokeColor(String color) {
+   }
+
+   /**
+    * Sets the fill color for all shapes drawn
+    * 
+    * @param color CSS color string to fill any shape. Defaults to "none", and 
+    *              setting it to "none" ensures that any shapes are not filled.
```

Comment:
It might make more sense to allow students to set `null` here instead. Or perhaps we create a second function that doesn't take an input. Java doesn't have default values.

Reply:
Agreed, I think a second function is the clearest pattern.